library/src/main/java/com/novoda/downloadmanager/NotificationCreator.java

```diff
@@ -33,10 +37,22 @@ public int getId() {
 
             @Override
             public Notification getNotification() {
-                NotificationCompat.Builder builder = new NotificationCompat.Builder(applicationContext, notificationChannelProvider.channelId());
+                int newNotificationId = getId();
+                NotificationCustomizer.NotificationDisplayState newNotificationDisplayState = notificationDisplayState();
+
+                if (builder == null || hasChanged(newNotificationId, newNotificationDisplayState)) {
```

Comment:
Now we create a new `Builder` whenever there isn't one present OR whenever the `id` and `displayState` has changed. If the display state and notification Id has not changed then there probably is no need to create a new builder. 

🤔 in fact, there might be no point in calling down to the `notificationCustomizer` if the `builder` has not changed between calls 🤔 What do you think? ❓

Reply:
well, you still need to call the `notificationCustomizer` in any case, because id and displayState might have not changed, but the payload could, isn't it?

Reply:
Yea you are completely correct 😄 just occurred to me too after reading it again 😂